### PR TITLE
Ensure User Password in ADAL throws the correct exception when passwo…

### PR DESCRIPTION
**Pull request by bgavrilMS**
…rd is wrong

### CODE REVIEW

File: adal/src/Microsoft.IdentityModel.Clients.ActiveDirectory/AuthenticationParameters.cs [hunk 1]
```diff
@@ -147,14 +148,14 @@ private static async Task<AuthenticationParameters> CreateFromResourceUrlCommonA
             try
             {
                 IHttpClient request = new HttpClientWrapper(resourceUrl.AbsoluteUri, null);
-                using (await request.GetResponseAsync().ConfigureAwait(false))
```
Comment (bgavrilMS) [inline adal/src/Microsoft.IdentityModel.Clients.ActiveDirectory/AuthenticationParameters.cs:150]: Minor refactoring - IHttpWebResponse does not need to implement IDisposable because it has nothing to dispose of.

File: adal/src/Microsoft.IdentityModel.Clients.ActiveDirectory/Internal/Http/HttpClientWrapper.cs [hunk 2]
```diff
@@ -141,7 +142,7 @@ public async Task<IHttpWebResponse> GetResponseAsync()
                         string.Format(CultureInfo.CurrentCulture,
                             "Response status code does not indicate success: {0} ({1}).",
                             (int) webResponse.StatusCode, webResponse.StatusCode),
-                        new AdalException(webResponse.ResponseString)));
```
Comment (bgavrilMS) [inline adal/src/Microsoft.IdentityModel.Clients.ActiveDirectory/Internal/Http/HttpClientWrapper.cs:144]: Minor refactoring: consolidate HttpResponse (MSAL) and HttpWebResponse (ADAL) under a single interface IWebResponse (in core).

File: adal/src/Microsoft.IdentityModel.Clients.ActiveDirectory/Microsoft.IdentityModel.Clients.ActiveDirectory.csproj [hunk 3]
```diff
@@ -93,8 +95,8 @@
     <Compile Include="Platforms\net45\**\*.cs" />
     <Compile Include="Features\WinCommon\**\*.cs" />
     <Compile Include="Features\CertificateAssertion\**\*.cs" />
-    <Compile Include="..\..\..\core\src\Platforms\net45\**\*.cs" LinkBase="Core\Platforms" />
-    <Compile Include="..\..\..\core\src\Features\Cache\**\*.cs" LinkBase="Core\Features" />
+    <Compile Include="..\..\..\core\src\Platforms\net45\**\*.cs" LinkBase="Core\Platforms\net45" />
```
Comment (bgavrilMS) [inline adal/src/Microsoft.IdentityModel.Clients.ActiveDirectory/Microsoft.IdentityModel.Clients.ActiveDirectory.csproj:98]: Unrelated fix - ensure the core files are included in correct directories in ADAL and MSAL. This change is cosmetic only.

**Comment by bgavrilMS** [comment]
@jennyf19 - Please merge this pull request yourself before adding more logic for username password, it will make things easier. The base branch is bogavril/iwa3

File: adal/src/Microsoft.IdentityModel.Clients.ActiveDirectory/AuthenticationParameters.cs (hunk 1 above)
Comment (henrik-me) [reply adal/src/Microsoft.IdentityModel.Clients.ActiveDirectory/AuthenticationParameters.cs:150]: Does it contain any objects which does? E.g. the HttpClient?

---
In reply to: [217285444](https://github.com/AzureAD/azure-activedirectory-library-for-dotnet/pull/1255#discussion_r217285444) [](ancestors = 217285444)

File: core/src/Exceptions/ExceptionDetail.cs [hunk 4]
```diff
@@ -50,5 +53,8 @@ internal class ExceptionDetail
         /// Raw response body received from the server.
         /// </summary>
         public string ResponseBody { get; set; }
+
+        public IDictionary<string, string> HttpHeaders { get; set; }
```
Comment (henrik-me) [inline core/src/Exceptions/ExceptionDetail.cs:57]: nit: missing comments

File: core/src/Exceptions/ExceptionDetail.cs [hunk 5]
```diff
@@ -26,6 +26,9 @@
 //------------------------------------------------------------------------------
 
 
+using System.Collections.Generic;
+using System.Net.Http.Headers;
+
 namespace Microsoft.Identity.Core
 {
     internal class ExceptionDetail
```
Comment (henrik-me) [inline core/src/Exceptions/ExceptionDetail.cs:34]: nit: class comments ?

File: core/src/Http/HttpRequest.cs [hunk 6]
```diff
@@ -61,12 +61,30 @@ private HttpRequest()
                         .ConfigureAwait(false);
         }
 
-        public static async Task<HttpResponse> SendGetAsync(Uri endpoint, Dictionary<string, string> headers,
+        public static async Task<HttpResponse> SendGetAsync(
+            Uri endpoint, 
+            Dictionary<string, string> headers,
             RequestContext requestContext)
```
Comment (henrik-me) [inline core/src/Http/HttpRequest.cs:67]: nit: why not one line? which tool are people using to do this?

**Review by henrik-me — CHANGES_REQUESTED** [review]
:clock1:

**Comment by henrik-me** [comment]
this thing needs lots of testing before being merged to dev, however if you are simply merging to the PR for u/p, I'm fine

**Comment by jennyf19** [comment]
sounds good...thanks.

---
In reply to: [421400448](https://github.com/AzureAD/azure-activedirectory-library-for-dotnet/pull/1255#issuecomment-421400448) [](ancestors = 421400448)

**Comment by jennyf19** [comment]
yes and yes

---
In reply to: [421424720](https://github.com/AzureAD/azure-activedirectory-library-for-dotnet/pull/1255#issuecomment-421424720) [](ancestors = 421424720)

**Review by henrik-me — APPROVED** [review]
